### ISSUE TITLE
[ Suggestion ] WavyElytra

### ISSUE THREAD

**Issue by SzczurekYT** [tr7zw/WaveyCapes#22]
Hello
I really like your mod I I tought that adding this same functionality for elytra would be could. Just imagine the waving wings. They also aren't static slab.
What do you think about that? If you would approve that maybe I could try to make this work and open PR?

**Comment by tr7zw** [comment]
![grafik](https://user-images.githubusercontent.com/5133936/176947085-2a36c1c3-f04e-4e0e-8518-09a01b88c87b.png)
I guess this answers it. I have thought about it, but it's a bit tricky(especially in the scope I am/was planning). My idea was to split the wings into parts, keep the outer edge at the vanilla fixed location, but let the inner edge be simulated. This has a "few" challenges: 
- split the wings into parts that still render correctly(that's where the mod is currently stuck, cause the wings render a bit more tricky than the cape)
- be able to animate the inner and outer vertices independently of each other so they can be attached to the stick animation
- update the stick animation to be able to handle this correctly(changed directions of gravity/that when flying slowly the elytra doesn't fold into itself)

But yea, if you have the experience to work on points of this, be my guest(here on Github/Discord).

**Comment by tr7zw** [comment]
![](https://images.unsplash.com/photo-1629100393327-a3a1550354fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8YmF0JTIwd2luZ3N8ZW58MHx8MHx8&w=1000&q=80)

Basically animated it in a style how the bone part outside is static(where the elytra have the thick edge), but animated on the thin inner part.

**Comment by SzczurekYT** [comment]
Cool idea, but for me it also seems complicated, I think that you are more experianced than me.
Also I'm not sure if minecraft elytra is big enough to make this effect you are proposing visible.
For me cape's physic applied to elytra would be enough.

**Comment by tr7zw** [comment]
Just blindly applying the cape physics would look really odd when it's just flopping around you while you are "gliding" with it.

**Comment by SzczurekYT** [comment]
Yeah, some isGliding check would fix that.

**Comment by tr7zw** [comment]
I mean to have it when not gliding is like 90% of the way there. So might as well add slight waves to it then.

**Comment by SzczurekYT** [comment]
Exactly, the gliding part is less important because  it looks fine in vanilla. With this mod it's just so odd for me that when I walk elytra has no movement.

**Comment by tr7zw** [comment]
What I meant with that is, when you can move and animate it while walking, then you might as well also animate it while flying. The issue is getting to this point at all.

**Comment by SzczurekYT** [comment]
Ah ok

**Comment by Aces-and-Jacks** [comment]
Unsure about how far you are with this, but as a temporary solution you can do something like this https://www.curseforge.com/minecraft/mc-mods/elytra-physics

Only a suggestion tho, if you don't want to do it that's fine as well

**Comment by tr7zw** [comment]
That actually looks like a nice solution for when walking. I had more in flight/ or start/stop of flight in mind. The recently added Simulation V2, which runs in 3d internally, might help with this. Still would require redoing the entire elytra model to be animate-able.

**Comment by DelofJ** [comment]
So you could let the animation when walking part to Elytra Physics and you do the flying animation part ?

**Comment by Aces-and-Jacks** [comment]
I'm guessing this is still on pause

**Comment by Mi6kbuttface** [comment]
I still need and want this. Elytra Physics "allows" modded elytras support but it is near impossible for a regular person to figure out how to do so.
